### DISCUSSION TITLE
remote editing

### DISCUSSION THREAD

**Discussion by cbertelli** [orbitalquark/textadept#253] [General]
Textmate2 and SublimeText may edit a file remotely using
https://github.com/aurora/rmate
Could it be possible tu use a similar solution in Textadept?
Vim shows another possible solution, using netrw.
Is someone already working on one of these?

**Comment by orbitalquark** [comment]
Not that I am aware of. Textadept does not have a built-in capability to connect to the internet or a local network. It is just a text editor.

**Reply by snoopy** [reply]
You could use [SSHFS](https://en.wikipedia.org/wiki/SSHFS). It lets you mount a remote filesystem locally which you can then edit normally with Textadept.  
I've used it in the past and it worked really well.

**Comment by orbitalquark** [comment]
Ah that's a good point. I've done that myself in the past. Thanks for the reminder!

**Comment by cbertelli** [comment]
Thank you very much. I'm already doing this, but working with several
projects I'm continuously mounting/unmounting remotes by rclone
(wonderful!).
What I thought of is adapting netrw (https://vimhelp.org/pi_netrw.txt.html)
to supplement textadept file explorer.
I understand the point by orbitalquark, pure KISS applied. Anyway it would
be useful to open a remote file on my desktop as rmate does. Sorry for
bothering everyone here.

**Comment by orbitalquark** [comment]
Not a bother at all. Discussions are good :)

**Comment by cbertelli** [comment]
Probably not worth while, but I have tried to understand if it is feasible. So I list here what I have found if useful for anyone or for myself-in-the-future (who knows?). This is the original page for rmate: blog.macromates.com/2011/mate-and-rmate/. There is a repository with python code for SublimeText that may be translated into lua and adapted to the file explorer of textadept: github.com/gja/sublime-rmate/. A similar approach was used for Atom here: atom.io/packages/remote-atom (even if coffee to lua is not so easy).
It's very convenient. You are logged in ssh to a host, you can rmate your file and it opens in your editor-of-choice on your desktop...

**Comment by rgieseke** [comment]
Should be possible to build an `rmate` extension for Textadept, i haven't checked the details how the VS Code or Atom etc. extensions work in detail but see for example the [Textadept Language Server](https://github.com/orbitalquark/textadept-lsp) implementation for an extension connecting to an external tool.

**Reply by cbertelli** [reply]
Hello, I didn't know, but [Weihang Ding](https://github.com/xiaozhuai) made [rmate-server](https://github.com/xiaozhuai/rmate-server), a generic rmate-server daemon that can broker a remote file to Textadept. It needs a configuration for any editor, mine for Textadept on MacOS is here:
```
{
    "host": "127.0.0.1",
    "port": 52698,
    "editor": {
        "command": "/Applications/Textadept.app/Contents/MacOS/textadept_osx",
        "args": [
            "{{file}}"
        ]
    }
}
```
Just change the command reference to the location of textadept on Linux or Windows, if you want to use it.
The developer suggested using the -w parameter for other editors
```
{
    "host": "127.0.0.1",
    "port": 52698,
    "editor": {
        "command": "mate",
        "args": [
            "-w",
            "{{file}}"
        ]
    }
}
```
but I found it to be unnecessary on Textadept.

**Reply by rgieseke** [reply]
Awesome, great find!
